### PR TITLE
add more "just do it" option to splunk start

### PR DESCRIPTION
**Pull request by arnej27959**
@bjormel please review and merge

### CODE REVIEW

**Comment by bjormel** [comment]
👍  
Thanks @arnej27959 
Just to make sure that this is reviewed by a someone that is fluent in the language: @vekterli

File: logforwarder/src/apps/vespa-logforwarder-start/child-handler.cpp [hunk 1]
```diff
@@ -15,13 +17,17 @@ ChildHandler::ChildHandler() : _childRunning(false) {}
...
-    LOG(debug, "starting splunk forwarder with command: '%s' '%s' '%s'",
-        argv[0], argv[1], argv[2]);
+    args.insert(args.begin(), path.c_str());
+    std::string dbg = "";
+    for (const char *arg : args) {
+        dbg.append(" '");
+        dbg.append(arg);
+        dbg.append("'");
+    }
+    LOG(debug, "starting splunk forwarder with command: %s", dbg.c_str());
```
Comment (vekterli) [inline logforwarder/src/apps/vespa-logforwarder-start/child-handler.cpp]: Could wrap this in `LOG_WOULD_LOG`, but this isn't exactly a hot inner loop so doesn't matter in practice

File: logforwarder/src/apps/vespa-logforwarder-start/child-handler.cpp [hunk 2]
```diff
@@ -33,10 +39,10 @@ runSplunk(const vespalib::string &prefix, const char *a1, const char *a2 = 0)
         char *cenv = const_cast<char *>(env.c_str()); // safe cast
         putenv(cenv);
         LOG(debug, "added to environment: '%s'", cenv);
-        char **cargv = const_cast<char **>(argv); // safe cast
-        execv(argv[0], cargv);
+        char **cargv = const_cast<char **>(args.data()); // safe cast
```
Comment (vekterli) [inline logforwarder/src/apps/vespa-logforwarder-start/child-handler.cpp]: Missing final `nullptr` entry in `args` vector to signal end of argument list?

**Review by vekterli — COMMENTED** [review]
See question re: argument list termination for `execv`, otherwise looks good.

File: logforwarder/src/apps/vespa-logforwarder-start/child-handler.cpp (hunk 2 above)
Comment (arnej27959) [reply logforwarder/src/apps/vespa-logforwarder-start/child-handler.cpp]: > Missing final nullptr entry in args vector to signal end of argument list?

yes, thank you!

**Comment by arnej27959** [comment]
@vekterli please take another look now

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**